xkcd.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+)
+
+const xkcdURL = "https://xkcd.com/%d/info.0.json"
+
+func GetComic(comicNum int) (Comic, error) {
+	resp, err := http.Get(fmt.Sprintf(xkcdURL, comicNum))
+	if err != nil {
+		return Comic{}, fmt.Errorf("failed to GET comic %d: %s", comicNum, err)
+	}
+	defer resp.Body.Close()
+
+	var comicRetrieved Comic
+	if err := json.NewDecoder(resp.Body).Decode(&comicRetrieved); err != nil {
+		return Comic{}, fmt.Errorf("failed to GET comic %d: %s", comicNum, err)
+	}
+
+	return comicRetrieved, nil
+}
+
+func GetLatestComicNumber() (int, error) {
+	resp, err := http.Get("https://xkcd.com/info.0.json")
+	if err != nil {
+		return 0, fmt.Errorf("failed to perform GET request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	var latest LatestComic
+	if err := json.NewDecoder(resp.Body).Decode(&latest); err != nil {
+		return 0, fmt.Errorf("failed to GET the latest comic number %d: %s", latest.Num, err)
```

Comment:
The error message isn't actually related to the issue, would be better to say "failed to unmarshall..."